pygments/lexers/thingsdb.py

```diff
@@ -39,21 +39,24 @@ class ThingsDBLexer(RegexLexer):
             (r'([-+]?0x[0-9a-fA-F]+)', Number.Hex),
             (r'[-+]?[0-9]+', Number.Integer),
             (r'[-+]?((inf|nan)([^0-9A-Za-z_]|$)|[0-9]*\.[0-9]+(e[+-][0-9]+)?)',
-            Number.Float),
+             Number.Float),
 
             # strings
             (r'(?:"(?:[^"]*)")+', String.Double),
             (r"(?:'(?:[^']*)')+", String.Single),
+            (r"(?:`(?:[^`]*)`)+", String.Backtick),
 
             # literals
             (r'(true|false|nil)\b', Keyword.Constant),
 
+            # name constants
+            (r'(FULL|USER|GRANT|CHANGE|JOIN|RUN|QUERY|'
+             r'DEBUG|INFO|WARNING|ERROR|CRITICAL|'
+             r'NO_IDS|INT_MIN|INT_MAX)\b', Name.Constant),
+
             # regular expressions
             (r'(/[^/\\]*(?:\\.[^/\\]*)*/i?)', String.Regex),
 
-            # thing id's
-            (r'#[0-9]+', Comment.Preproc),
```

Comment:
Could you explain why this is removed?

Reply:
Retrieving things using the hash # syntax was only available in old 0.x versions of ThingsDB and was later removed as part of the language. A simple function call which is more generec, thing(ID), must be used instead.